api/migrations/0003_fake_data.py

```diff
@@ -152,29 +190,86 @@ def _make_teacher_under_18_recipe(self):
             is_validated=self.faker.pybool,
         )
 
-    def _make_fake_coordinators(self):
-        """Makes fake coordinators."""
-        self.coordinator_recipe.make(_quantity=AMOUNT_OF_COORDINATORS)
+    def __get_random_time_or_none(self) -> time | None:
+        return self.faker.random_element([None, self.faker.time_object()])
 
-    def _make_fake_students(self):
-        """Makes fake students."""
-        self.student_recipe.make(_quantity=AMOUNT_OF_STUDENTS)
+    def _make_group_recipe(self) -> Recipe:
+        group_common_recipe = self._make_group_common_recipe(
+            APP_NAME + ".Group", teachers_min_value=1, teachers_max_value=3
+        )
+        return group_common_recipe.extend(
+            availability_slot=lambda: self._get_random_amount_of_objects(
+                DayAndTimeSlot, min_length=10, max_length=20
+            ),
+            is_for_staff_only=self.faker.pybool,
+            language_and_level=lambda: self.faker.random_element(
+                LanguageAndLevel.objects.all()
+            ),
+            lesson_duration_in_minutes=lambda: self.faker.pyint(
+                min_value=30, max_value=120, step=30
+            ),
+            status=lambda: self.faker.random_element(GroupStatus.values),
+            start_date=self.faker.past_date,
+            end_date=self.faker.future_date,
+            # FIXME: Sometimes all of the days are generated as None.
+            #  It violates the model's constraint.
+            #  Just restart the migration and pray to the God of Random.
+            monday=self.__get_random_time_or_none,
+            tuesday=self.__get_random_time_or_none,
+            wednesday=self.__get_random_time_or_none,
+            thursday=self.__get_random_time_or_none,
+            friday=self.__get_random_time_or_none,
+            saturday=self.__get_random_time_or_none,
+            sunday=self.__get_random_time_or_none,
```

Comment:
actually no group can have classes more often than 3 times a week, so if a group has 7 times all filled in, it will be an unrealistic group that might look unrealistic on the dashboard.

Maybe generate a dictionary with 2-3 days randomly chosen as keys and then do `extend` with `(**dict)`?

Reply:
Problem with generating the dictionary of fixed days is that then we will have same days for all groups, it's bad even if we have different time for them. I didn't manage to find any option to create this dict randomly because we can use lambda only as a field value, not as a field itself. I have an idea of doing check after recipe "making", but it definitely will look more ugly than current solution. 

So I decided to keep current approach - generate for all days but with `random_time_or_none`, it usually means that groups will have 3-4 active days and 3-4 `None` days. 

Also made a fix to be sure that needed amount of groups is created, check it out

Reply:
We could also try to use weighting (https://faker.readthedocs.io/en/master/providers/baseprovider.html#faker.providers.BaseProvider.random_elements)
 here for choosing between None or time, but I'm not sure if it's needed